x/mint/types/inflations.go

```diff
@@ -0,0 +1,68 @@
+package types
+
+import (
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// NewInflation returns a new Inflation object with the given denom, target_address
+// and inflation_rate
+func NewInflation(denom string, targetAddress string, inflationRate sdk.Dec) Inflation {
+	return Inflation{
+		Denom:         denom,
+		TargetAddress: targetAddress,
+		InflationRate: inflationRate,
+	}
+}
+
+func CalculateInflationPerBlock(inflation *Inflation, blocksPerYear uint64) (result sdk.Dec, err error) {
+	inflationPerBlockPlusOne, err := inflation.InflationRate.Add(sdk.OneDec()).ApproxRoot(blocksPerYear)
+	if err != nil {
+		panic(err)
+	}
+	inflationPerBlock := inflationPerBlockPlusOne.Sub(sdk.OneDec())
+	return inflationPerBlock, nil
+}
+
+func CalculateInflationNewCoins(inflationPerBlock sdk.Dec, supply sdk.Coin) (result sdk.Coins) {
+	newCoinAmounts := (inflationPerBlock.MulInt(supply.Amount)).TruncateInt()
+	return sdk.NewCoins(sdk.NewCoin(supply.Denom, newCoinAmounts))
+}
+
+// ValidateInflation ensures validity of Inflation object fields
+// TODO(JS): potentially to introduce negative inflations
+func ValidateInflation(inflation Inflation) error {
+	if inflation.InflationRate.IsNegative() {
```

Comment:
I think we should not restrict inflation, and enable the **full** range of mathematicaly valid inflation values `-1 >= inflation`